### PR TITLE
Implement Client functions.

### PR DESCRIPTION
**Pull request by humdrum**
**What this PR does / why we need it**:
- Implement full features of the client.
- Add Integration test for the client and the document.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
- Due to CI status. All the tests related to the client have been skipped.
- ~~The auto-sync feature isn't working correctly.~~

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: Sources/Core/Client.swift [hunk 1]
```diff
@@ -305,6 +241,381 @@ final class Client {
...
+        }
+
+        do {
+            try await withThrowingTaskGroup(of: Void.self) { group in
+                for (key, attachment) in self.attachmentMap where attachment.isRealtimeSync {
+                    let docChanged = await attachment.doc.hasLocalChanges()
+
+                    if docChanged || attachment.remoteChangeEventReceived ?? false {
+                        self.attachmentMap[key]?.remoteChangeEventReceived = false
+                        try await self.syncInternal(attachment.doc)
```
[the remark this answers is not in the document]
Comment (humdrum) [reply Sources/Core/Client.swift:439]: That's exactly what I want. Thanks.

File: Sources/Core/Client.swift [hunk 2]
```diff
@@ -305,6 +242,394 @@ final class Client {
...
+
+        do {
+            try await withThrowingTaskGroup(of: Void.self) { group in
+                documents.forEach { document in
+                    group.addTask {
+                        try await self.syncInternal(document)
+                    }
+                }
+
+                for try await _ in group {}
```
Comment (hyeongsik-won) [inline Sources/Core/Client.swift:348]: This code seems unnecessary for similar reasons as [this](https://github.com/yorkie-team/yorkie-ios-sdk/pull/31#discussion_r1016122446).

Comment (humdrum) [reply Sources/Core/Client.swift:348]: That's for exception catch.

**Review by hyeongsik-won — APPROVED** [review]
LGTM!

File: Sources/Core/Client.swift (hunk 2 above)
Comment (humdrum) [reply Sources/Core/Client.swift:348]: I found a better function. waitForAll()

**Pull request merged**